extensions/amazon-aws/src/secrets.tsx

```diff
@@ -0,0 +1,125 @@
+  SecretsManagerClient,
+  ListSecretsCommand,
+  GetSecretValueCommand,
+  SecretListEntry,
+} from "@aws-sdk/client-secrets-manager";
+import { Action, ActionPanel, Icon, List } from "@raycast/api";
+import { useCachedPromise } from "@raycast/utils";
+import { useCallback, useState } from "react";
+import AWSProfileDropdown, { AWS_URL_BASE } from "./aws-profile-dropdown";
+
+export default function Secrets() {
+  const [search, setSearch] = useState<string>("");
+  const [showValue, setShowValue] = useState<boolean>(false);
+  const { data: secrets, error, isLoading, revalidate } = useCachedPromise(fetchSecrets, [search]);
+
+  const showDetailFn = useCallback(() => setShowValue(!showValue), [showValue]);
+
+  return (
+    <List
+      isLoading={isLoading}
+      isShowingDetail={showValue}
+      searchBarPlaceholder="Filter secrets by name..."
+      searchBarAccessory={<AWSProfileDropdown onProfileSelected={revalidate} />}
+      onSearchTextChange={setSearch}
+      throttle
+    >
+      {search.length < 4 ? (
+        <List.EmptyView
+          title="Missing Search Query"
+          icon={Icon.MagnifyingGlass}
+          description="The search will begin when at least 4 characters are provided."
+        />
+      ) : error ? (
+        <List.EmptyView title={error.name} description={error.message} icon={Icon.Warning} />
+      ) : (
+        secrets?.map((secret) => (
+          <Secret key={secret.Name} secret={secret} showValue={showValue} setShowValue={showDetailFn} />
+        ))
+      )}
+    </List>
+  );
+}
+
+function Secret({
+  secret,
+  showValue,
+  setShowValue,
+}: {
+  secret: SecretListEntry;
+  showValue: boolean;
+  setShowValue: () => void;
+}) {
+  const { data: secretDetails } = useCachedPromise(fetchSecret, [secret.ARN]);
+  return (
+    <List.Item
+      id={secret.Name || ""}
+      icon={Icon.Bookmark}
+      title={secret.Name || ""}
+      detail={
+        <List.Item.Detail
+          markdown={secretDetails?.SecretString || secretDetails?.SecretBinary?.toString() || ""}
+          metadata={
+            <List.Item.Detail.Metadata>
+              <List.Item.Detail.Metadata.Label title="Description" text={secret.Description} />
+              <List.Item.Detail.Metadata.Label title="ARN" text={secret.ARN} />
+              <List.Item.Detail.Metadata.Separator />
+              <List.Item.Detail.Metadata.Label title="Creation Date" text={secret.CreatedDate?.toLocaleString()} />
+              <List.Item.Detail.Metadata.Label title="Last Accessed" text={secret.LastAccessedDate?.toLocaleString()} />
+              <List.Item.Detail.Metadata.Separator />
+              <List.Item.Detail.Metadata.Label title="Rotation" text={`${secret.RotationEnabled ? "true" : "false"}`} />
+            </List.Item.Detail.Metadata>
+          }
+        />
+      }
+      actions={
+        <ActionPanel>
+          <Action.SubmitForm title={showValue ? "Hide Value" : "Show Value"} onSubmit={() => setShowValue()} />
```

Comment:
I'm aware other commands have used the `SubmitForm` action for such a use case, but a simple [Action](https://developers.raycast.com/api-reference/user-interface/actions#action) is probably a better choice.

```suggestion
          <Action title={showValue ? "Hide Value" : "Show Value"} onSubmit={() => setShowValue()} />
```

Reply:
Fixed